src/vinyldns/serdes.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2018 Comcast Cable Communications Management, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add module docstring."""
+import re
+
+camel_pat = re.compile(r'([A-Z])')
+under_pat = re.compile(r'_([a-z])')
+
+
+def camel_to_underscore(name):
+    """
+    Given a string in camelCase, convert to snake_case
+    :param name: The string to be converted
+    :return: A String formatted in snake_case
+    """
+    return camel_pat.sub(lambda x: '_' + x.group(1).lower(), name)
+
+
+def underscore_to_camel(name):
+    """
+    Given a string in snake_case, convert to camelCase
+    :param name: The string to be converted
+    :return: A String formatted in camelCase
+    """
+    return under_pat.sub(lambda x: x.group(1).upper(), name)
+
+
+def to_dict(obj, cls=None):
+    """
+    Given an object, serializes all fields to a dictionary including nested objects.
+
+    Converts the field names to camelCase along the way suitable for convention in the API
+    :param obj: An object instance
+    :param cls: The type of class, useful for recursion
+    :return: A fully populated dictionary with field names converted to camelCase
+    """
+    if isinstance(obj, dict):
+        data = {}
+        for (k, v) in obj.items():
+            # Runs a conversion of snake_case to camelCase to be compliant with the API
+            data[underscore_to_camel(k)] = to_dict(v, cls)
+        return data
+    elif hasattr(obj, "_ast"):
+        return to_dict(obj._ast())
+    elif hasattr(obj, "__iter__") and not isinstance(obj, str):
+        return [to_dict(v, cls) for v in obj]
+    elif hasattr(obj, "__dict__"):
+        data = dict([(underscore_to_camel(key), to_dict(value, cls))
+                     for key, value in obj.__dict__.items()
```

Comment:
Indentation seems a bit funky here.

Reply:
The `for` is aligned in the list comprehension under the `[`.  It does look funky but it is correct